### PR TITLE
Set default $MAX_MEMORY

### PR DESCRIPTION
**Pull request by lincolnthree**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/628/
Test PASSed.

File: dist/src/main/resources/bin/windup [hunk 1]
```diff
@@ -165,6 +165,10 @@ if [ -r "$WINDUP_HOME/addons/" ] ; then
   ADDONS_DIR="--immutableAddonDir $WINDUP_HOME/addons/"
 fi
 
+if [ -z "$MAX_MEMORY" ] ; then
+  MAX_MEMORY="2048m"
+fi
+
```
Comment (OndraZizka) [inline dist/src/main/resources/bin/windup]: Is that enough? I see JVM taking over 3 GB total. MAX_MEMORY goes to heap I guess? So the total could be around those 3 GB?

Anyway, machines with 4 GB could have around 4 GB of swap, so I think we could safely set 3 GB heap. Risk of swapping other processes out of RAM is better than risking performance degradation due to GC running all the time and evt. hitting OoMEx. my2c.

Comment (jsight) [inline dist/src/main/resources/bin/windup]: I'm still -1 on this patch (for reasons that I have mentioned previously), but I definitely wouldn't recommend going larger. Swapping and GC'ed languages combine extremely poorly for various reasons, and there are still a lot of 4GB languages out there.

**Comment by lincolnthree** [comment]
Agreed.

**Pull request closed without merging**